ecosystem/indexer-grpc/indexer-grpc-cache-worker/src/worker.rs

```diff
@@ -205,90 +211,74 @@ async fn process_transactions_from_node_response(
         },
         Response::Data(data) => {
             let transaction_len = data.transactions.len();
-            let first_transaction = data
-                .transactions
-                .first()
-                .context("There were unexpectedly no transactions in the response")?;
-            let last_transaction = data
-                .transactions
-                .last()
-                .context("There were unexpectedly no transactions in the response")?;
-            let start_version = first_transaction.version;
-            let first_transaction_pb_timestamp = first_transaction.timestamp.clone();
-            let last_transaction_pb_timestamp = last_transaction.timestamp.clone();
-
-            log_grpc_step(
-                SERVICE_TYPE,
-                IndexerGrpcStep::CacheWorkerReceivedTxns,
-                Some(start_version as i64),
-                Some(last_transaction.version as i64),
-                first_transaction_pb_timestamp.as_ref(),
-                last_transaction_pb_timestamp.as_ref(),
-                Some(batch_start_time.elapsed().as_secs_f64()),
-                Some(size_in_bytes),
-                Some((last_transaction.version - first_transaction.version + 1) as i64),
-                None,
-            );
+            let mut cache_operator_clone = cache_operator.clone();
+            let task: JoinHandle<anyhow::Result<()>> = tokio::spawn({
+                let first_transaction = data
+                    .transactions
+                    .first()
+                    .context("There were unexpectedly no transactions in the response")?;
+                let first_transaction_version = first_transaction.version;
+                let last_transaction = data
+                    .transactions
+                    .last()
+                    .context("There were unexpectedly no transactions in the response")?;
+                let last_transaction_version = last_transaction.version;
+                let start_version = first_transaction.version;
+                let first_transaction_pb_timestamp = first_transaction.timestamp.clone();
+                let last_transaction_pb_timestamp = last_transaction.timestamp.clone();
+
+                log_grpc_step(
+                    SERVICE_TYPE,
+                    IndexerGrpcStep::CacheWorkerReceivedTxns,
+                    Some(start_version as i64),
+                    Some(last_transaction_version as i64),
+                    first_transaction_pb_timestamp.as_ref(),
+                    last_transaction_pb_timestamp.as_ref(),
+                    Some(batch_start_time.elapsed().as_secs_f64()),
```

Comment:
`batch_start_time` gets reset after each batch. can we reset it before each download? we want to measure the ony the download duration here